### PR TITLE
[Intl] handle null date and time types

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29705
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php [hunk 1]
```diff
@@ -118,13 +118,13 @@ class IntlDateFormatter
...
-     * @param int    $calendar Calendar to use for formatting or parsing. The only currently
-     *                         supported value is IntlDateFormatter::GREGORIAN (or null using the default calendar, i.e. "GREGORIAN")
-     * @param string $pattern  Optional pattern to use when formatting
+     * @param string   $locale   The locale code. The only currently supported locale is "en" (or null using the default locale, i.e. "en")
+     * @param int|null $datetype Type of date formatting, one of the format type constants
+     * @param int|null $timetype Type of time formatting, one of the format type constants
+     * @param mixed    $timezone Timezone identifier
+     * @param int      $calendar Calendar to use for formatting or parsing. The only currently
+     *                           supported value is IntlDateFormatter::GREGORIAN (or null using the default calendar, i.e. "GREGORIAN")
+     * @param string   $pattern  Optional pattern to use when formatting
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php]: also `string|null` + see `setPattern`

**Review by ro0NL — COMMENTED** [review]
`getDateType` + `static create()` also need a patch

**Comment by xabbuh** [comment]
@ro0NL I fail to see which change is needed for `getDateType()`.

**Review by ro0NL — APPROVED** [review]
never mind, $dateType cant be null anymore  :)

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**